plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1613,6 +1614,11 @@ public static function get_subscription_list_data() {
 			if ( ! empty( $updates[ $subscription['product_id'] ] ) ) {
 				$subscription['version'] = $updates[ $subscription['product_id'] ]['version'];
 			}
+
+			// If the update endpoint returns a URL, we prefer it over the default PluginURI.
+			if ( ! empty( $updates[ $subscription['product_id'] ]['url'] ) ) {
+				$subscription['product_url'] = $updates[ $subscription['product_id'] ]['url'];
+			}
```

Comment:
This works, but overwriting the previously declared value complicates this code a bit. It becomes harder to track how the value was set.

I'm curious if you considered removing this second `foreach` loop, as it seems redundant given it iterates the same subscriptions that are defined in the loop above?

Reply:
Hi @denho, the code is a bit unobvious, but the function combines data from 2 sources: connected subscriptions and locally installed Woo products. The connected subscriptions [are fetched here](https://github.com/woocommerce/woocommerce/pull/47950/files#diff-32b26bf8430b272079f3c981317024f2a7d7848b4198ae7878716cc133d10961R1567), and then locally installed Woo products [are added here](https://github.com/woocommerce/woocommerce/pull/47950/files#diff-32b26bf8430b272079f3c981317024f2a7d7848b4198ae7878716cc133d10961R1587). 

The [second loop](https://github.com/woocommerce/woocommerce/pull/47950/files#diff-32b26bf8430b272079f3c981317024f2a7d7848b4198ae7878716cc133d10961R1614) iterates over this merged subscriptions list, and further refines it. I couldn't find a way to make this clearer in code, so I added some (hopefully) descriptive comments.